src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -324,7 +324,17 @@ public void focusGained(final FocusEvent e) {
     m_rightHandSidePanel.add(m_smallView, BorderLayout.NORTH);
     m_tabsPanel.setBorder(null);
     m_rightHandSidePanel.add(m_tabsPanel, BorderLayout.CENTER);
-    m_actionButtons = new ActionButtons(m_data, m_mapPanel, this);
+
+    MovePanel movePanel = new MovePanel(m_data, m_mapPanel, this);
+    m_actionButtons = new ActionButtons(m_data, m_mapPanel, movePanel, this);
+
+    List<KeyListener> keyListeners = ImmutableList.of(this.getArrowKeyListener(), movePanel.getUndoMoveKeyListener());
```

Comment:
Same as another comment, consistently buggy at least as the 'n' key functionality. Fixing the obviously unnecessary code depends on #305. Again, since this is marked as a beta feature and suffers from an existing bug, I would like to move forward.